src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.SolutionCreator.cs

```diff
@@ -356,6 +361,14 @@ private async Task<Project> UpdateProjectInfoAsync(Project project, Checksum inf
                 var oldMap = await GetDocumentMapAsync(existingTextDocumentStates, olds.Object).ConfigureAwait(false);
                 var newMap = await GetDocumentMapAsync(_assetProvider, news.Object).ConfigureAwait(false);
 
+                // If more than two documents changed during a single update, perform a bulk synchronization on the
```

Comment:
looks fine with me.  do we have any explicit tests where we resync with multiple files changing just to ensure that nothing breaks with this?

Reply:
I'll try to make one